string[] name
string[] state # Armed, Disconnected, or Error
string[] error